src/inventory/InventoryHelpersTrait.php

```diff
@@ -144,6 +144,17 @@ public function addItem(Item ...$slots) : array{
 			}
 		}
 
+		/** @var Item[] $returnSlots */
+		$returnSlots = [];
+
+		foreach($itemSlots as $item){
+			$returnSlots[] = $this->internalAddItem($item);
```

Comment:
This should check if the returned item `isNull()`, and don't add it to the result if so.